src/NumericalAlgorithms/Interpolation/InterpolationTargetReceiveVars.hpp

```diff
@@ -61,13 +95,20 @@ void callback_and_cleanup(
       db::RemoveTags<>, db::AddSimpleTags<>,
       db::AddComputeTags<
           typename InterpolationTargetTag::compute_items_on_target>>(*box);
-  InterpolationTargetTag::post_interpolation_callback::apply(new_box, *cache,
-                                                             temporal_id);
 
-  // In the future, when generalizing this function to accommodate
-  // horizon finding, we will need to think more carefully about
-  // control flow and the point at which the temporal_id should be
-  // popped from the list.
+  // apply_callback should return true if we are done with this
```

Comment:
We aren't sure what you are trying to do, but be aware that db::create_from creates aliases to items in the original box (so shared pointer copies).  So the values from compute items in the new_box will depend upon the order of instructions in the function apply_callback and afterwards.  So we don't think this is what you want.  db::create_copy creates a genuine copy (time-intensive) of the original box.  Another  possibility is to move box into create_from to get a new non-const box and later remove the compute items which is super cheap . (ask @nilsdeppe for details if this is not clear)